crypto/fipsmodule/ec/ec_key.c

```diff
@@ -381,6 +381,32 @@ int EC_KEY_check_fips(const EC_KEY *key) {
     return 0;
   }
 
+  // Check that the coordinates are within the range [0,p-1], when the (raw)
+  // point is affine; i.e. Z=1.
+  // This is the case when validating a received public key.
+  // Note: The check for x and y being negative seems superfluous since
+  // ec_felem_to_bignum() calls BN_bin2bn() which sets the `neg` flag to 0.
+  if(ec_felem_equal(key->pub_key->group, &key->pub_key->group->one, &key->pub_key->raw.Z)) {
+    BIGNUM *x = BN_new();
+    BIGNUM *y = BN_new();
+    int ret = 1;
+    if (key->pub_key->group->meth->felem_to_bytes == 0 ||
+        !ec_felem_to_bignum(key->pub_key->group, x, &key->pub_key->raw.X) ||
+        !ec_felem_to_bignum(key->pub_key->group, y, &key->pub_key->raw.Y) ||
+        BN_is_negative(x) ||
+        BN_is_negative(y) ||
+        BN_cmp(x, &key->pub_key->group->field) >= 0 ||
+        BN_cmp(y, &key->pub_key->group->field) >= 0) {
+      OPENSSL_PUT_ERROR(EC, EC_R_COORDINATES_OUT_OF_RANGE);
```

Comment:
Nit Pick: 
There are a few ways that we can get here and not all of them imply that the Coordinates are out of range ( if there is an issue with the group methods or if there is an allocation failure from the ec to bignum operation we will end up throwing a coordinates out of range error, and this might complicate troubleshooting).

Reply:
Done. Thanks for the help